lib/crypto.js

```diff
@@ -282,7 +282,16 @@ Sign.prototype.sign = function sign(options, encoding) {
 
   var key = options.key || options;
   var passphrase = options.passphrase || null;
-  var ret = this._handle.sign(toBuf(key), null, passphrase, options);
+
+  // Options specific to RSA
+  var rsaPadding = options.padding || constants.RSA_PKCS1_PADDING;
+  var pssSaltLength = constants.RSA_PSS_SALTLEN_MAX_SIGN;
+  if (typeof options.saltLength === 'number') {
+    pssSaltLength = options.saltLength;
+  }
```

Comment:
I guess it is reasonable to only accept integers. Invalid integers should be rejected by OpenSSL, that's how `publicEncrypt` etc. handle it. Something along `if((options.padding & 0xffffffff) === options.padding)`? This is the first thing that came to my mind, I don't know of any functions to distinguish between numbers and integers.